### ISSUE TITLE
Getting: 502 Bad Gateway -- nginx

### ISSUE THREAD

**Issue by cbhiii** [hassio-addons/addon-grocy#197]
# Problem/Motivation

> Grocy not loading in HA

## Expected behavior

> Grocy to open and land on default start screen

## Actual behavior

> Getting error message: `502 Bad Gateway -- nginx`

## Steps to reproduce

> Just tried to open Grocy after not using it for awhile. I rebooted the host. I then removed it and reinstalled it via supervisior. Not sure what other steps to try.

Log file from the app:

```
-----------------------------------------------------------
 Add-on: Grocy
 ERP beyond your fridge! A groceries & household management solution for your home
-----------------------------------------------------------
 Add-on version: 0.14.0
 You are running the latest version of this add-on.
 System: Home Assistant OS 6.1  (aarch64 / raspberrypi3-64)
 Home Assistant Core: 2021.6.6
 Home Assistant Supervisor: 2021.06.6
-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums or the Discord chat.
-----------------------------------------------------------
[cont-init.d] 00-banner.sh: exited 0.
[cont-init.d] 01-log-level.sh: executing... 
[cont-init.d] 01-log-level.sh: exited 0.
[cont-init.d] grocy.sh: executing... 
[11:03:43] INFO: Patching Grocy to fix relative URL handling...
patching file views/layout/default.blade.php
[cont-init.d] grocy.sh: exited 0.
[cont-init.d] nginx.sh: executing... 
[cont-init.d] nginx.sh: exited 0.
[cont-init.d] php-fpm.sh: executing... 
[cont-init.d] php-fpm.sh: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[11:03:45] INFO: Starting PHP-FPM...
[11:03:57] INFO: Starting NGinx....
2021/07/02 11:04:27 [error] 545#545: *1 recv() failed (104: Connection reset by peer) while reading response header from upstream, client: 172.30.32.2, server: a0d7b954-grocy, request: "GET / HTTP/1.1", upstream: "fastcgi://127.0.0.1:9002", host: "192.168.32.6:8123", referrer: "http://192.168.32.6:8123/hassio/addon/a0d7b954_grocy/info"

 (cleared browser cache and restarted Chrome in between these two tries listed here)

2021/07/02 11:08:13 [error] 545#545: *3 recv() failed (104: Connection reset by peer) while reading response header from upstream, client: 172.30.32.2, server: a0d7b954-grocy, request: "GET / HTTP/1.1", upstream: "fastcgi://127.0.0.1:9002", host: "192.168.32.6:8123", referrer: "http://192.168.32.6:8123/hassio"
```

![Pasted Graphic 12](https://user-images.githubusercontent.com/10853381/124299757-caf40b00-db2b-11eb-81c8-b85030108397.png)

**Comment by bcampru** [comment]
I am having the same error

**Comment by jef-meike** [comment]
I also have the same error.

**Comment by coehler200** [comment]
I have the same issue. Glad this seems to be an issue that more people have. Maybe now it will get fixed. So far no help on the Discord.

**Comment by frenck** [comment]
Please, "I have the same issue", is rarely helpful with debugging issues.

For example, share system information, so the common things between the installations can be found and thus finding them common cause.

Thanks

**Comment by javiermarasco** [comment]
I just installed the add on and have the same problem, here is my log:

'''
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] 01-grocy: applying... 
[fix-attrs.d] 01-grocy: exited 0.
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] 00-banner.sh: executing... 
-----------------------------------------------------------
 Add-on: Grocy
 ERP beyond your fridge! A groceries & household management solution for your home
-----------------------------------------------------------
 Add-on version: 0.14.0
 You are running the latest version of this add-on.
 System: Home Assistant OS 6.1  (aarch64 / raspberrypi3-64)
 Home Assistant Core: 2021.6.6
 Home Assistant Supervisor: 2021.06.8
-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums or the Discord chat.
-----------------------------------------------------------
[cont-init.d] 00-banner.sh: exited 0.
[cont-init.d] 01-log-level.sh: executing... 
Log level is set to INFO
[cont-init.d] 01-log-level.sh: exited 0.
[cont-init.d] grocy.sh: executing... 
[22:17:45] INFO: Patching Grocy to fix relative URL handling...
patching file views/layout/default.blade.php
[cont-init.d] grocy.sh: exited 0.
[cont-init.d] nginx.sh: executing... 
[cont-init.d] nginx.sh: exited 0.
[cont-init.d] php-fpm.sh: executing... 
[cont-init.d] php-fpm.sh: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[22:17:47] INFO: Starting PHP-FPM...
[22:17:56] INFO: Starting NGinx....
2021/07/06 22:23:26 [error] 607#607: *1 recv() failed (104: Connection reset by peer) while reading response header from upstream, client: 172.30.32.2, server: a0d7b954-grocy, request: "GET / HTTP/1.1", upstream: "fastcgi://127.0.0.1:9002", host: "192.168.XXX.XXX:8123", referrer: "http://192.168.XXX.XXX:8123/a0d7b954_grocy" '''

**Comment by coehler200** [comment]
My logs are already on the Discord, but I'll post them here as well for your reference. Also just noticed that all of us are running a rpi3 maybe that has something to do with it?
`[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] 01-grocy: applying... 
[fix-attrs.d] 01-grocy: exited 0.
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] 00-banner.sh: executing... 
-----------------------------------------------------------
 Add-on: Grocy
 ERP beyond your fridge! A groceries & household management solution for your home
-----------------------------------------------------------
 Add-on version: 0.14.0
 You are running the latest version of this add-on.
 System: Home Assistant OS 6.1  (aarch64 / raspberrypi3-64)
 Home Assistant Core: 2021.6.6
 Home Assistant Supervisor: 2021.06.8
-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums or the Discord chat.
-----------------------------------------------------------
[cont-init.d] 00-banner.sh: exited 0.
[cont-init.d] 01-log-level.sh: executing... 
[cont-init.d] 01-log-level.sh: exited 0.
[cont-init.d] grocy.sh: executing... 
[15:50:50] INFO: Patching Grocy to fix relative URL handling...
patching file views/layout/default.blade.php
[cont-init.d] grocy.sh: exited 0.
[cont-init.d] nginx.sh: executing... 
[cont-init.d] nginx.sh: exited 0.
[cont-init.d] php-fpm.sh: executing... 
[cont-init.d] php-fpm.sh: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[15:50:52] INFO: Starting PHP-FPM...
[15:51:03] INFO: Starting NGinx....
2021/07/06 15:51:15 [error] 606#606: *1 recv() failed (104: Connection reset by peer) while reading response header from upstream, client: 172.30.32.2, server: a0d7b954-grocy, request: "GET / HTTP/1.1", upstream: "fastcgi://127.0.0.1:9002", host: "subdomain.duckdns.org", referrer: "https://subdomain.duckdns.org/a0d7b954_grocy"`

**Comment by SalvaRomo** [comment]
Frenck: found this reference on plesk in case you have access to the add on grocy server  configuration file:
https://support.plesk.com/hc/en-us/articles/213903705
Hope it helps,

**Comment by herrtschaefer** [comment]
Same issue here:
log: `[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] 01-grocy: applying... 
[fix-attrs.d] 01-grocy: exited 0.
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] 00-banner.sh: executing... 
-----------------------------------------------------------
 Add-on: Grocy
 ERP beyond your fridge! A groceries & household management solution for your home
-----------------------------------------------------------
 Add-on version: 0.14.0
 You are running the latest version of this add-on.
 System: Home Assistant OS 6.1  (aarch64 / raspberrypi3-64)
 Home Assistant Core: 2021.7.1
 Home Assistant Supervisor: 2021.06.8
-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums or the Discord chat.
-----------------------------------------------------------
[cont-init.d] 00-banner.sh: exited 0.
[cont-init.d] 01-log-level.sh: executing... 
[cont-init.d] 01-log-level.sh: exited 0.
[cont-init.d] grocy.sh: executing... 
[17:37:13] INFO: Patching Grocy to fix relative URL handling...
patching file views/layout/default.blade.php
[cont-init.d] grocy.sh: exited 0.
[cont-init.d] nginx.sh: executing... 
[cont-init.d] nginx.sh: exited 0.
[cont-init.d] php-fpm.sh: executing... 
[cont-init.d] php-fpm.sh: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[17:37:16] INFO: Starting PHP-FPM...
[17:37:29] INFO: Starting NGinx....
2021/07/11 17:39:18 [error] 552#552: *1 recv() failed (104: Connection reset by peer) while reading response header from upstream, client: 172.30.32.2, server: a0d7b954-grocy, request: "GET / HTTP/1.1", upstream: "fastcgi://127.0.0.1:9002", host: "192.168.0.73:8123", referrer: "http://192.168.0.73:8123/a0d7b954_grocy"`

**Comment by coehler200** [comment]
@SalvaRomo any idea on how I'd access that? I assume I'd have to get into the container and edit that file.

**Comment by coehler200** [comment]
Alright so using Portainer I've tried looking for /etc/sw-cp-server/config Well unless I'm misunderstanding the location of sw-cp-server it doesn't appear to be on the container.

**Comment by frenck** [comment]
It is, unfortunately, an upstream problem:

<https://gitlab.alpinelinux.org/alpine/aports/-/issues/12692>

Caused by a libxml2 upgrade done in:

<https://gitlab.alpinelinux.org/alpine/aports/-/merge_requests/21695>

**Comment by frenck** [comment]
Resolved on dev with the recent upgrades for the upcoming Grocy 3.10.

**Comment by deb42** [comment]
Should this issue had been resolved with v0.15.0 because update to Grocy 3.10. was part of it? 

Im am currently still getting this problem:

`[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] 01-grocy: applying... 
[fix-attrs.d] 01-grocy: exited 0.
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] 00-banner.sh: executing... 
-----------------------------------------------------------
 Add-on: Grocy
 ERP beyond your fridge! A groceries & household management solution for your home
-----------------------------------------------------------
 Add-on version: 0.15.0
 You are running the latest version of this add-on.
 System: Home Assistant OS 6.2  (aarch64 / raspberrypi3-64)
 Home Assistant Core: 2021.7.4
 Home Assistant Supervisor: 2021.06.8
-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums or the Discord chat.
-----------------------------------------------------------
[cont-init.d] 00-banner.sh: exited 0.
[cont-init.d] 01-log-level.sh: executing... 
[cont-init.d] 01-log-level.sh: exited 0.
[cont-init.d] grocy.sh: executing... 
[12:14:43] INFO: Patching Grocy to fix relative URL handling...
patching file views/layout/default.blade.php
Hunk #1 succeeded at 88 with fuzz 2 (offset 1 line).
[cont-init.d] grocy.sh: exited 0.
[cont-init.d] nginx.sh: executing... 
[cont-init.d] nginx.sh: exited 0.
[cont-init.d] php-fpm.sh: executing... 
[cont-init.d] php-fpm.sh: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[12:14:46] INFO: Starting PHP-FPM...
[12:15:03] INFO: Starting NGinx....
2021/08/04 12:16:39 [error] 617#617: *1 recv() failed (104: Connection reset by peer) while reading response header from upstream, client: 172.30.32.2, server: a0d7b954-grocy, request: "GET / HTTP/1.1", upstream: "fastcgi://127.0.0.1:9002", host: "deb42-hass.duckdns.org", referrer: "https://deb42-hass.duckdns.org/hassio"`

**Comment by jrrcor** [comment]
Same here. As this been resolved?
